libs/geometry/src/SurfaceOrientation.cpp

```diff
@@ -226,6 +227,7 @@ SurfaceOrientation* OrientationBuilderImpl::buildWithUvs() {
     memset(tan2.data(), 0, sizeof(float3) * vertexCount);
     for (size_t a = 0; a < triangleCount; ++a) {
         uint3 tri = triangles16 ? uint3(triangles16[a]) : triangles32[a];
+        assert_invariant(tri.x < vertexCount && tri.y < vertexCount && tri.z < vertexCount);
```

Comment:
So is this checking for possible bugs in our code?

If this assert can fail due to something the client does, this should use the `PANIC` stuff and the method should not be `noexcept`. 

If this code is not directly called by our client, but rather used by filament itself, then `assert_invariant` is fine -- it would just mean that the caller (i.e. us) did something wrong, in which case we need to make sure this doesn't happen.

Reply:
This checks for badly formed model data.  This is a bounds check that I think should not occur in release builds since it's in a tight loop.  I just want to make it easier to catch this issue using a debug build without the help of ASAN.

The client directly calls this method, but since I didn't want to add a bounds check to release builds, I felt that an assert was the way to go.